homeassistant/components/weatherflow_cloud/sensor.py

```diff
@@ -0,0 +1,239 @@
+    ),
+    WeatherFlowCloudSensorEntityDescription(
+        key="sea_level_pressure",
+        translation_key="sea_level_pressure",
+        value_fn=lambda data: data.observation.obs[0].sea_level_pressure,
+        native_unit_of_measurement=UnitOfPressure.MBAR,
+        device_class=SensorDeviceClass.ATMOSPHERIC_PRESSURE,
+        state_class=SensorStateClass.MEASUREMENT,
+        suggested_display_precision=3,
+    ),
+    WeatherFlowCloudSensorEntityDescription(
+        key="station_pressure",
+        translation_key="station_pressure",
+        value_fn=lambda data: data.observation.obs[0].station_pressure,
+        native_unit_of_measurement=UnitOfPressure.MBAR,
+        device_class=SensorDeviceClass.ATMOSPHERIC_PRESSURE,
+        state_class=SensorStateClass.MEASUREMENT,
+        suggested_display_precision=3,
+    ),
+    # Lightning Sensors
+    WeatherFlowCloudSensorEntityDescription(
+        key="lightning_strike_count",
+        translation_key="lightning_strike_count",
+        state_class=SensorStateClass.MEASUREMENT,
+        value_fn=lambda data: data.observation.obs[0].lightning_strike_count,
+    ),
+    WeatherFlowCloudSensorEntityDescription(
+        key="lightning_strike_count_last_1hr",
+        translation_key="lightning_strike_count_last_1hr",
+        state_class=SensorStateClass.MEASUREMENT,
+        value_fn=lambda data: data.observation.obs[0].lightning_strike_count_last_1hr,
+    ),
+    WeatherFlowCloudSensorEntityDescription(
+        key="lightning_strike_count_last_3hr",
+        translation_key="lightning_strike_count_last_3hr",
+        state_class=SensorStateClass.MEASUREMENT,
+        value_fn=lambda data: data.observation.obs[0].lightning_strike_count_last_3hr,
+    ),
+    WeatherFlowCloudSensorEntityDescription(
+        key="lightning_strike_last_distance",
+        translation_key="lightning_strike_last_distance",
+        state_class=SensorStateClass.MEASUREMENT,
+        device_class=SensorDeviceClass.DISTANCE,
+        native_unit_of_measurement=UnitOfLength.KILOMETERS,
+        value_fn=lambda data: data.observation.obs[0].lightning_strike_last_distance,
+    ),
+    WeatherFlowCloudSensorEntityDescription(
+        key="lightning_strike_last_epoch",
+        translation_key="lightning_strike_last_epoch",
+        device_class=SensorDeviceClass.TIMESTAMP,
+        value_fn=lambda data: datetime.fromtimestamp(
+            data.observation.obs[0].lightning_strike_last_epoch, tz=UTC
+        ),
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up WeatherFlow sensors based on a config entry."""
+
+    coordinator: WeatherFlowCloudDataUpdateCoordinator = hass.data[DOMAIN][
+        entry.entry_id
+    ]
+
+    stations = coordinator.data.keys()
+
+    for sensor_description in WF_SENSORS:
+        for station_id in stations:
+            async_add_entities(
```

Comment:
Turn it around by the way, we should call `async_add_entities` as less as possible as it is expensive. So make a generator expression and just puteverything in async_add_entities at once